src/NuGetGallery.Core/GitHub/GitHubCacheTransformer.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NuGetGallery.GitHub
+{
+    public class GitHubCacheTransformer
+    {
+        public static Dictionary<string, NuGetPackageGitHubInformation> GetNuGetPackagesDependents(IReadOnlyList<RepositoryInformation> repositories)
+        {
+            var dependentsPerPackage = new Dictionary<string, List<RepositoryInformation>>(StringComparer.InvariantCultureIgnoreCase);
+            foreach (var repo in repositories)
+            {
+                foreach (var dependency in repo.Dependencies)
+                {
+                    List<RepositoryInformation> packageDependents = null;
+                    if (dependentsPerPackage.ContainsKey(dependency))
```

Comment:
Consider something like:

```suggestion
                    if (!dependentsPerPackage.TryGetValue(dependency, out var packageDependents))
                    {
                        packageDependents = new List<RepositoryInformation>();
                        dependentsPerPackage.Add(packageDependents);
                    }
```